src/PIL/ImageOps.py

```diff
@@ -136,28 +136,101 @@ def autocontrast(image, cutoff=0, ignore=None):
     return _lut(image, lut)
 
 
-def colorize(image, black, white):
+def colorize(image, black, white, mid=None, blackpoint=0,
+             whitepoint=255, midpoint=127):
     """
-    Colorize grayscale image.  The **black** and **white**
-    arguments should be RGB tuples; this function calculates a color
-    wedge mapping all black pixels in the source image to the first
-    color, and all white pixels to the second color.
+    Colorize grayscale image.
+    This function calculates a color wedge which maps all black pixels in
+    the source image to the first color and all white pixels to the
+    second color. If **mid** is specified, it uses three-color mapping.
+    The **black** and **white** arguments should be RGB tuples or color names;
+    optionally you can use three-color mapping by also specifying **mid**.
+    Mapping positions for any of the colors can be specified
+    (e.g. **blackpoint**), where these parameters are the integer
+    value in [0, 255] corresponding to where the corresponding color
+    should be mapped.
 
     :param image: The image to colorize.
     :param black: The color to use for black input pixels.
     :param white: The color to use for white input pixels.
+    :param mid: The color to use for midtone input pixels.
+    :param blackpoint: an int value [0, 255] for the black mapping.
+    :param whitepoint: an int value [0, 255] for the white mapping.
+    :param midpoint: an int value [0, 255] for the midtone mapping.
     :return: An image.
     """
+
+    # Initial asserts
     assert image.mode == "L"
+    assert 0 <= whitepoint <= 255
+    assert 0 <= blackpoint <= 255
+    assert 0 <= midpoint <= 255
+    assert blackpoint <= whitepoint
```

Comment:
It's adding assumptions, but perhaps `assert 0 <= blackpoint <= midpoint <= whitepoint <= 255` instead of these four? But only if those assumptions are correct.

And if so, the following asserts aren't needed.

Reply:
see other comments on asserts. new commit coming soon.